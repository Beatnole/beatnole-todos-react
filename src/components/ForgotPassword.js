import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { firebase } from '../firebase/firebase'
import { history } from '../routers/AppRouter'

const PasswordForgetPage = () =>
    <div className="box-layout">
        <div className="box-layout__box">  
                <h1>Password Forget</h1>
                <PasswordForgetForm />
        </div>
    </div>
const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

const INITIAL_STATE = {
  email: '',
  error: null,
};

class PasswordForgetForm extends Component {
  constructor(props) {
    super(props);

    this.state = { ...INITIAL_STATE };
  }

  onSubmit = (event) => {
    const { email } = this.state;

    firebase.auth().sendPasswordResetEmail(email)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        history.push('/email')
      })
      .catch(error => {
        this.setState(byPropKey('error', error));
      });

    event.preventDefault();
  }

  render() {
    const {
      email,
      error,
    } = this.state;

    const isInvalid = email === '';

    return (
      <form onSubmit={this.onSubmit}>
        <input
          autoFocus
          className="text-input__email"
          value={this.state.email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type="text"
          placeholder="Email Address"
        />
        <button disabled={isInvalid} className={isInvalid ? "button" : "button button-password-reset"} type="submit">
          Send Reset Email
        </button>

        { error && <p>{error.message}</p> }
      </form>
    );
  }
}

const PasswordForgetLink = () =>
  <p>
    <Link to={routes.PASSWORD_FORGET}>Forgot Password?</Link>
  </p>

export default PasswordForgetPage;

export {
  PasswordForgetForm,
  PasswordForgetLink,
};