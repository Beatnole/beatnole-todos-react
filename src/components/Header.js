import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'
import { startSignOut } from '../actions/auth'
import { userInfo, userPhoto } from '../app'

export const Header = ({ startSignOut }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Todo's</h1>
                </Link>
                <button className="button button--link" onClick={startSignOut}>Logout</button>
            </div>
            
            <div className="content-container__image ">
                <div className="header__content--user_container">
                        <span className="header__title header__title--userinfo">{userInfo}</span>
                        <img className="user-profile-image" src={userPhoto ? userPhoto : '/images/blue.ico'}/>
                </div>
            </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startSignOut: () => dispatch(startSignOut())
})

export default connect(undefined, mapDispatchToProps)(Header)