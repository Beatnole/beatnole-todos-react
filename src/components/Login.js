import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { startLogin } from '../actions/auth'

export const Login = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Todo Login</h1>
            <button onClick={startLogin} className="button button_adjust button-group">Login with Google</button>
            <Link className="button button_adjust button-group" to="/email">Login with Email</Link>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(Login)