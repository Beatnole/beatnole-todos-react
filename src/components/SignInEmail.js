import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { firebase } from '../firebase/firebase'
import { Link } from 'react-router-dom'

const SignInPage = ({ history }) =>
  <div className="box-layout">
    <div className="box-layout__box">
            <h1>Todo Sign In</h1>
            <SignInForm history={history} />
            <Link className="button button-password-reset" to="/forgotEmail">Forgot Password</Link>
        </div>
  </div>

  const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

const INITIAL_STATE = {
  email: '',
  password: '',
  error: null,
};

class SignInForm extends Component {
  constructor(props) {
    super(props);

    this.state = { ...INITIAL_STATE };
  }

  onSubmit = (event) => {
    const {
      email,
      password,
    } = this.state;

    const {
      history,
    } = this.props;

    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        history.push('/dashboard')
      })
      .catch(error => {
        this.setState(byPropKey('error', error));
      });

    event.preventDefault();
  }

  render() {
    const {
      email,
      password,
      error,
    } = this.state;

    const isInvalid =
      password === '' ||
      email === '';

    return (
      <div>
          <form onSubmit={this.onSubmit} >
            <input
              autoFocus
              className="text-input__email"
              value={email}
              onChange={event => this.setState(byPropKey('email', event.target.value))}
              type="text"
              placeholder="Email Address"
            />
            <input
              className="text-input__email"
              value={password}
              onChange={event => this.setState(byPropKey('password', event.target.value))}
              type="password"
              placeholder="Password"
            />
            <button disabled={isInvalid}
            type="submit" 
            className={isInvalid ? "button button--secondary" : "button"} >
              Sign In
            </button>

            { error && <p>{error.message}</p> }
          </form>
      </div>
    );
  }
}

export default withRouter(SignInPage);

export {
  SignInForm,
};