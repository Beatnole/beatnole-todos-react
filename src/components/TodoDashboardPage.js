import React from 'react';
import TodoList from './TodoList'
import TodoListFilters from './TodoListFilters'
import TodoSummary from './TodoSummary'
import AddTodo from './AddTodoPage'

const TodoDashboardPage = () => (
    <div>
        <TodoSummary />
        <TodoListFilters />
        <TodoList />
        <AddTodo />
    </div>
)

export default TodoDashboardPage