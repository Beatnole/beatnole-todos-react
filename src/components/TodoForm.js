import React from 'react'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'


export default class TodoForm extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            description: props.todo ? props.todo.description : '',
            createdAt: props.todo ? moment(props.todo.createdAt) : moment(),
            calendarFocused: false,
            error: ''
    }
}
    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() => ({ description }) ) 
    }
    
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }) )
        }
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }) )
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (!this.state.description) {
            this.setState(() => ({ error: 'Please provide a description' }) )
        } else {
            this.setState(() => ({ error: '' }) )
            this.props.onSubmit({
                description: this.state.description,
                createdAt: this.state.createdAt.valueOf()
            })
        }
        this.state.description = ''
    }

    render() {
        return (
                <form className="form" onSubmit={this.onSubmit}>
                  {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <input
                        type="text"
                        placeholder="Add Todo"
                        autoFocus
                        className="text-input"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <SingleDatePicker 
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false }
                    />
                   
                        <button className="button">{this.props.todo ? 'Save' : 'Add'} Todo</button>
                    
                </form>
        )
    }
}