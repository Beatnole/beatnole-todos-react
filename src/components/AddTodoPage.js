import React from 'react';
import { connect } from 'react-redux'
import TodoForm from './TodoForm'
import { startAddTodo } from '../actions/todos'

export class AddTodoPage extends React.Component {
    onSubmit = (todo) => {
        this.props.startAddTodo(todo)
    }
    render() {
        return (
        <div>
            <div>
                <div className="input-group__item">
                    <h1 className="page-header__title"></h1>    
                </div>
            </div>
            <div className="content-container">
                <TodoForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        </div>
    )}
}

const mapDispatchToProps = (dispatch) => {
    return {
        startAddTodo: (todo) => dispatch(startAddTodo(todo))
    }
}

export default connect(undefined, mapDispatchToProps)(AddTodoPage)