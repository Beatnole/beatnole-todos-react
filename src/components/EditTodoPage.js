import React from 'react';
import { connect } from 'react-redux';
import TodoForm from './TodoForm'
import { startEditTodo, startRemoveTodo } from '../actions/todos'

export class EditTodoPage extends React.Component {
    onSubmit = (todo) => {
        this.props.startEditTodo(this.props.todo.id, todo)
        this.props.history.push('/')
    }
    onClick = () => {
        this.props.startRemoveTodo({ id: this.props.todo.id })
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                    <div className="page-header">
                        <div className="content-container">
                            <h1 className="page-header__title">Edit Todo</h1>    
                        </div>
                    </div>
                    <div className="content-container">
                    <TodoForm
                        todo={this.props.todo}
                                onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onClick}
                    >Remove Todo</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        todo: state.todos.find((todo) => todo.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditTodo: (id, todo) => dispatch(startEditTodo(id, todo)),
        startRemoveTodo: (id) => dispatch(startRemoveTodo(id))
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(EditTodoPage)