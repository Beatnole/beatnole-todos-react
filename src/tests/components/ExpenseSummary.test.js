import React from 'react';
import { ExpensesSummary } from '../../components/ExpenseSummary'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import { filters, altFilters } from '../fixtures/filters';


test('ExpenseSummary Display with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={5013} />)
    
    expect(wrapper).toMatchSnapshot();
  });

  test('ExpenseSummary Display with 23 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={124525013} />)
    
    expect(wrapper).toMatchSnapshot();  
});


