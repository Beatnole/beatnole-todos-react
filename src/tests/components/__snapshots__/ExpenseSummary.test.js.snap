// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary Display with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       of 
      <span />
       totalling 
      <span>
        $50.13
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`ExpenseSummary Display with 23 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       of 
      <span />
       totalling 
      <span>
        $1,245,250.13
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
